frontend/src/pages/SignupPage.js

```diff
@@ -23,6 +23,12 @@ function SignupPage() {
     const [dialogMsg, setDialogMsg] = useState("");
     const [isSignupSuccess, setIsSignupSuccess] = useState(false);
 
+    useEffect(() => {

```

Comment:
it's the other way round, when the user is logged in, he/she should not navigate to the signup and login page, thus it will redirect back to the home page